components/access-management/kfam/bindings.go

```diff
@@ -73,6 +73,30 @@ func getBindingName(binding *Binding) (string, error) {
 	return reg.ReplaceAllString(nameRaw, "-"), nil
 }
 
+func getSecurityCondition(binding *Binding, userIdHeader string, userIdPrefix string) []*istioSecurity.Condition {
+	if userIdHeader == "" {
+		return []*istioSecurity.Condition{}
+	}
```

Comment:
By default Istio flavored Kubeflow installation, there is anonymous user injection to any request which doesn't have UserID header set (https://github.com/kubeflow/manifests/blob/v1.2-branch/istio/add-anonymous-user-filter/base/kustomization.yaml). We aren't installing this filter, and we're explicitly denying request without UserID header.
And this code should cover it. According to [documentation](https://pkg.go.dev/istio.io/api/security/v1beta1#AuthorizationPolicy), not defining `Rules` in AuthorizationPolicy is equivalent to setting a default of deny for the target workloads.

Reply:
@subreptivus in this PR, we're just trying to port the existing functionality to AuthorizationPolicy.
Allowing all requests to a namespace if a header is not provided to KFAM is new behavior, a dangerous one too!
If you think this is needed, let's discuss in an issue. But for this PR, let's scope it down to just port what we already have to AuthorizationPolicy.

Reply:
Yeah, you're right. I've just checked.
First of all, there is check for the header/argument name, rather than actual value of the header. My bad, I've missed it, it wasn't my initial intention.
And second, you're right to. I thought if I'll add empty condition, it will be like following:
```yaml
spec:
 rules:
 - {}
 ```
 But in fact it produces the following:
 ```yaml
spec:
  rules:
  - when: []
 ```
 which is really not what I wanted to achieve. So this is settled.